assets/js/data.coffee

```diff
@@ -384,16 +460,25 @@ class Data
       cur = do cur.getParent
       if cur.is @viewRoot
         return answer
-      if cur.id == @root.id
+      if do cur.isRoot
         return null
       if @collapsed cur
         answer = cur
 
+  wouldBeCircularInsertTree: (row, parent) ->
+    # Precondition: tree is not already circular
+    # Rather than checking for each descendent, whether that descendent is an ancestor (old method)
+    # Instead, check if the row is an ancestor of the new parent, which is sufficient.
+    _.contains (@allAncestors parent.id, { inclusive: true }), row.id
```

Comment:
To explain further, if there WAS a clone underneath the row which was an ancestor of 'parent', then 'row' would also be an ancestor of 'parent'